src/common/utxobased/engine/utils.ts

```diff
@@ -202,12 +165,12 @@ export const deriveXprivFromKeys = (args: {
 }
 
 export const deriveXpubsFromKeys = (args: {
-  keys: UtxoKeyFormat
+  privateKey: PrivateKey
   coin: string
   network: NetworkEnum
 }): CurrencyFormatKeys => {
   const xpubs: CurrencyFormatKeys = {}
-  for (const format of getWalletSupportedFormats(args)) {
+  for (const format of asWalletKeysFormats(args.privateKey)) {
```

Comment:
`args.privateKey` is typed, and therefore should *not* be cleaned. Instead, do `getSupportedFormats(privateKey.format)`.